### DISCUSSION TITLE
[H3] Send a response but continue something afterwards.

### DISCUSSION THREAD

**Discussion by WerdoxDev** [unjs/community#42] [Q&A]
I'm using H3 and have a few routes that would be much faster If I could send the response sooner but still do some other work after.

An example is when a user sends a messages, the message is saved on the db and a response is sent. Then the message is also broadcasted with WebSocket but this happens after the message was sent so to not hang the request for an unrelated thing.

**Comment by pi0** [comment]
You can use `event.waitUntil` to do something like this:

```JS
  event.waitUntil(() => {
    // Runs in the background after the response
    // Broadcast a message to all clients
  })

  return "Hello, World!"
```

**Reply by WerdoxDev** [reply]
Hmm this would be perfect but it seems like it's a feature of nitro and not H3. Or maybe part of v2? I couldn't find any docs for it

**Reply by pi0** [reply]
Yes, as it integrates async promise with runtimes such as cloudflare workers runtime.
